moveit_ros/planning/moveit_cpp/src/planning_component.cpp

```diff
@@ -195,7 +199,70 @@ PlanningComponent::PlanSolution PlanningComponent::plan(const PlanRequestParamet
   //    visual_tools_->publishRobotState(last_solution_trajectory_->getLastWayPoint(), rviz_visual_tools::TRANSLUCENT);
   //  }
   //}
-  return *last_plan_solution_;
+  return *last_plan_solution_;  // TODO: Make better
+}
+
+PlanningComponent::PlanSolution
+PlanningComponent::plan(const PlanningComponent::MultiPipelinePlanRequestParameters& parameters,
+                        std::function<bool(std::vector<std::thread> planning_threads,
+                                           PlanningComponent::MultiPipelinePlanRequestParameters& parameters)>
+                            stop_criterion_callback,
```

Comment:
I'm guessing the idea is that this callback gets called periodically in a separate thread that monitors `planning_solutions` for changes?

Reply:
I still have to figure out a good way how it is possible to end a thread from outside. The way this is currently implemented the execution of `pipeline->generatePlan(planning_scene, req, res)`  takes the most time and should be interrupt-able. But to do this I'd need to check e.g. an atomic flag within generatePlan on a regular basis. From my understanding the cpp std::thread API does not allow killing/interruption a thread from outside without leaking memory :/ As you previously pointed out, maybe boost thread is a solution for this.

Reply:
Yes.